pkg/vadmin/alter_sc_vc.go

```diff
@@ -0,0 +1,95 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vadmin
+
+import (
+	"context"
+	"fmt"
+
+	vops "github.com/vertica/vcluster/vclusterops"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/altersc"
+)
+
+const (
+	primary   = "primary"
+	secondary = "secondary"
+)
+
+// AlterSubclusterType will promote/demote a subcluster in the vertica cluster.
+func (v *VClusterOps) AlterSubclusterType(ctx context.Context, opts ...altersc.Option) error {
+	v.setupForAPICall("AlterSubclusterType")
+	defer v.tearDownForAPICall()
+	v.Log.Info("Starting vcluster AlterSubclusterType")
+
+	// get the certs
+	certs, err := v.retrieveNMACerts(ctx)
```

Comment:
I remember this API doesn't need NMA certs. It only made HTTPs calls. You can remove lines 39 to 42.